File: AcroStack/Services/Dtos/Chat/ConversationDto.cs
using System;

namespace AcroStack.Services.Dtos.Chat;

/// <summary>A conversation preview shown in the contact/chat list.</summary>
public class ConversationDto
{
    /// <summary>IdentityUser.Id of the other party.</summary>
    public Guid TargetUserId { get; set; }

    public string TargetUserName { get; set; } = string.Empty;
    public string TargetName { get; set; } = string.Empty;
    public string TargetSurname { get; set; } = string.Empty;
    public string? TargetEmail { get; set; }

    public string LastMessage { get; set; } = string.Empty;

    public DateTime LastMessageDate { get; set; }

    public ChatMessageSideDto LastMessageSide { get; set; }

    public int UnreadMessageCount { get; set; }
}

File: react/src/api/models/acroStack/chat/ConversationDto.ts
/* oxlint-disable */

import type { AcroStackChatChatMessageSideDto } from "./ChatMessageSideDto";

/**
 * @description A conversation preview shown in the contact/chat list.
 * @type object
 */
export type AcroStackChatConversationDto = {
  /**
   * @description IdentityUser.Id of the other party.
   *
   * Format: `uuid`
   * @type string | undefined
   */
  targetUserId?: string;
  targetUserName?: string | null;
  targetName?: string | null;
  targetSurname?: string | null;
  targetEmail?: string | null;
  lastMessage?: string | null;
  /**
   * @description
   * Format: `date-time`
   * @type string | undefined
   */
  lastMessageDate?: string;
  /**
   * @description
   * Format: `int32`
   * @type integer | undefined
   */
  lastMessageSide?: AcroStackChatChatMessageSideDto;
  /**
   * @description
   * Format: `int32`
   * @type integer | undefined
   */
  unreadMessageCount?: number;
};
